serial/serial_resources/component_instance.py

```diff
@@ -0,0 +1,112 @@
+import serial
+from ..api_client import APIClient
+from .component_instance_link import ComponentInstanceLink
+
+class ComponentInstance:
+    def __init__(self, component_instance_data):
+        self.client = APIClient(serial.api_key, serial.base_url)
+        self.data = component_instance_data
+        self.links = [] 
+
+    def add_link(self, link_name, child_identifier, break_prior_links=False, process_entry=None):
+        """
```

Comment:
A lot of these documentation strings are out of date - should be cleaned up before sending them to production. Lesson learned here for me is to do that at the end after all redesigns & comments :)